model/lifecycle.go

```diff
@@ -524,17 +522,35 @@ type BuildCreateArgs struct {
 	SourceRev     string                  // githash of the revision that triggered this build
 	DefinitionID  string                  // definition ID of the trigger used to create this build
 	Aliases       ProjectAliases          // project aliases to use to filter tasks created
-	Session       mongo.SessionContext    // session context to use for transactions
 	DistroAliases distro.AliasLookupTable // map of distro aliases to names of distros
 }
 
 // CreateBuildFromVersion creates a build given all of the necessary information
 // from the corresponding version and project and a list of tasks.
 func CreateBuildFromVersion(args BuildCreateArgs) (string, error) {
+	b, tasks, err := CreateBuildFromVersionNoInsert(args)
+	if err != nil {
+		return "", err
+	}
+	if err = b.Insert(); err != nil {
+		return "", errors.Wrapf(err, "error inserting build %s", b.Id)
+	}
+	if err = tasks.InsertUnordered(context.Background()); err != nil {
```

Comment:
should this take a context passed in?

Reply:
Potentially, but I'd like to keep this signature the same for now so that we don't have to change a bunch of places